### PR TITLE
Feature/login page

### PR DESCRIPTION
**Pull request by Kadrian**
This adds a login page that posts the following to `POST /api/login`
```
{
  user: string,
  password: string
}
```

expects to receive
```
{
  access_token: string 
}
```

and a 4xx status in case of an error.

If there's an access token coming back, the user is redirected to the main page, the token is stored in the localStorage and reused.

The `?access_token` in the url overrides this token.

The environment parameter:
```
DISABLE_LOGIN=1 yarn start 
```
disables the login page, in case we want to test an instance without url parameter *and* without login page.

Looks like: 
Bildschirmfoto 2020-01-28 um 23 40 53

### CODE REVIEW

**Comment by awildturtok** [comment]
Du schwein! 🥳🥳🥳

**Comment by thoniTUB** [comment]
Habs mit meiner Backend Implementation getestet und es hat geklappt. Danke Kai!

File: frontend/lib/js/api/api.js [hunk 1]
```diff
@@ -211,3 +212,16 @@ export function postFilterValuesResolve(
 export function getMe(): Promise<GetMeResponseT> {
   return fetchJson(apiUrl() + `/me`);
 }
+
+export function postLogin(
+  user: string,
+  password: string
+): Promise<PostLoginResponseT> {
+  return fetchJsonUnauthorized(apiUrl() + "/auth", {
```
[the remark this answers is not in the document]
Comment (Kadrian) [reply frontend/lib/js/api/api.js]: Arr, true. I'll adjust it again

[the remark this answers is not in the document]
Comment (Kadrian) [reply frontend/lib/js/api/api.js]: Should work now. Can you check again?

[the remark this answers is not in the document]
Comment (Kadrian) [reply frontend/lib/js/api/api.js]: Good hint, should be fixed now as well!

**Pull request merged**